waiter/src/waiter/config.clj

```diff
@@ -0,0 +1,37 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns waiter.config)
+
+(def ^:private config-promise (promise))
+
+(defn initialize-config
+  "Initializes the config."
+  [config-value]
+  (if-not (realized? config-promise)
+    (deliver config-promise config-value)
```

Comment:
This operation is racy. Should we have a lock to ensure the `realized?` and `deliver` operations happen atomically?
Alternatively, we could `deliver` the value, and then require `(identical? @config-promise config-value)`. (If the contents are identical, do we really care about the double-initialization?)

Reply:
Done.